src/SubscriptionService/SubscriptionService.cs

```diff
@@ -0,0 +1,492 @@
+                                                                          CancellationToken cancellationToken)
+        {
+            PersistentSubscriptionSettingsBuilder settings = this.GetDefaultPersistentSubscriptionSettingsBuilder().StartFromBeginning();
+
+            await this.EventStoreConnection.CreatePersistentSubscriptionAsync(stream, groupName, settings, this.DefaultUserCredentials);
+        }
+
+        /// <summary>
+        /// Events the appeared.
+        /// </summary>
+        /// <param name="subscription">The subscription.</param>
+        /// <param name="resolvedEvent">The resolved event.</param>
+        /// <param name="subscriptionConfiguration">The subscription configuration.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        private async Task EventAppeared(EventStorePersistentSubscriptionBase subscription,
+                                         ResolvedEvent resolvedEvent,
+                                         Subscription subscriptionConfiguration,
+                                         CancellationToken cancellationToken)
+        {
+            try
+            {
+                //If instructed to, we will ignore event types beginning with the character $.
+                //This helps stop sending unused events to our read models etc (which will probably end up parked anyway)
+                if (resolvedEvent.Event == null)
+                {
+                    // This indicates we have a badly formatted event so just ignore it as nothing can be done :|
+                    subscription.Acknowledge(resolvedEvent);
+                    return;
+                }
+
+                this.Trace($"EventAppearedFromPersistentSubscription with Event Id {resolvedEvent.Event.EventId}");
+
+                String serialisedData = this.GetSerialisedDataFromEvent(resolvedEvent);
+
+                await this.PublishMessage(subscriptionConfiguration.EndPointUrl, serialisedData, resolvedEvent.Event.EventId, cancellationToken);
+
+                subscription.Acknowledge(resolvedEvent);
+            }
+            catch(Exception e)
+            {
+                this.Trace(e);
+                this.NakEvent(subscription, resolvedEvent, e);
+            }
+        }
+
+        /// <summary>
+        /// Handles the AuthenticationFailed event of the EventStoreConnection control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="ClientAuthenticationFailedEventArgs" /> instance containing the event data.</param>
+        private void EventStoreConnection_AuthenticationFailed(Object sender,
+                                                               ClientAuthenticationFailedEventArgs e)
+        {
+            this.Trace($"Connection {e.Connection.ConnectionName} AuthenticationFailed, Reason {e.Reason}");
```

Comment:
Do we need this?

Reply:
No as we are not worried about that, will remove